src/translation-service.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/**
+ * @module utils/translation-service
+ */
+
+const translations = {};
+
+/**
+ * Merges package translations to existing ones.
+ * These translations can be used later with {@link module:utils/translations-service~translate translate}
+ *
+ *		add( 'pl', {
+ *			'OK': 'OK',
+ *			'Cancel [context: reject]': 'Anuluj'
+ *		} );OK
+ *
+ * @param {String} lang
+ * @param {Object.<String, Object>} packageDictionary
+ * @returns undefined
+ */
+export function add( lang, packageDictionary ) {
+	if ( !( lang in translations ) ) {
+		translations[ lang ] = {};
+	}
+
+	const dictionary = translations[ lang ];
+
+	for ( const translationKey in packageDictionary ) {
+		const translation = packageDictionary[ translationKey ];
+		dictionary[ translationKey ] = translation;
+	}
+}
+
+/**
+ * Translates string if the translation of the string was previously added using {@link module:utils/translations-service~add add}.
+ * Otherwise returns original (English) sentence.
+ *
+ *		translate( 'pl', 'Cancel [context: reject]' );
+ *
+ * @param {String} lang Translation language.
+ * @param {String} translationKey Sentence which is going to be translated.
+ * @returns {String} Translated sentence.
+ */
+export function translate( lang, translationKey ) {
+	if ( !existTranslationKey( lang, translationKey ) ) {
```

Comment:
I'm not sure whether this code makes any sense (and if it does, it requires comments). Please explain me how's it going to work in the dev env (where no translations are defined and `t()` is called with the original strings) and in the situation when translation keys were replaced with chosen language. AFAICT, it's going to work in both of these situations, but I'd like to have confirmation and comments for that. Perhaps the best way for them will be the documentation of this function which now is incorrect ("returns original (English) sentence" is not entirely true in the second mentioned situation).

Please also remember to explain the not yet implemented support for multiple languages.